sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/SamplingRuleApplier.java

```diff
@@ -0,0 +1,248 @@
+      String name,
+      SpanKind spanKind,
+      Attributes attributes,
+      List<LinkData> parentLinks) {
+    return fixedRateSampler.shouldSample(
+        parentContext, traceId, name, spanKind, attributes, parentLinks);
+  }
+
+  @Nullable
+  private static String getArn(Attributes attributes, Resource resource) {
+    String arn = resource.getAttributes().get(ResourceAttributes.AWS_ECS_CONTAINER_ARN);
+    if (arn != null) {
+      return arn;
+    }
+    String cloudPlatform = resource.getAttributes().get(ResourceAttributes.CLOUD_PLATFORM);
+    if (cloudPlatform == null
+        || !cloudPlatform.equals(ResourceAttributes.CloudPlatformValues.AWS_LAMBDA)) {
+      return null;
+    }
+    arn = resource.getAttributes().get(ResourceAttributes.FAAS_ID);
+    if (arn != null) {
+      return arn;
+    }
+    return attributes.get(ResourceAttributes.FAAS_ID);
+  }
+
+  @Nullable
+  private static String getServiceType(Resource resource) {
+    String cloudPlatform = resource.getAttributes().get(ResourceAttributes.CLOUD_PLATFORM);
+    if (cloudPlatform == null) {
+      return null;
+    }
+    switch (cloudPlatform) {
+      case ResourceAttributes.CloudPlatformValues.AWS_EC2:
+        return "AWS::EC2::Instance";
+      case ResourceAttributes.CloudPlatformValues.AWS_ECS:
+        return "AWS::ECS::Container";
+      case ResourceAttributes.CloudPlatformValues.AWS_EKS:
+        return "AWS::EKS::Container";
+      case ResourceAttributes.CloudPlatformValues.AWS_ELASTIC_BEANSTALK:
+        return "AWS::ElasticBeanstalk::Environment";
+      case ResourceAttributes.CloudPlatformValues.AWS_LAMBDA:
+        return "AWS::Lambda::Function";
+      default:
+        return null;
+    }
+  }
+
+  private static Matcher toMatcher(String globPattern) {
+    if (globPattern.equals("*")) {
+      return TrueMatcher.INSTANCE;
+    }
+
+    for (int i = 0; i < globPattern.length(); i++) {
+      char c = globPattern.charAt(i);
+      if (c == '*' || c == '?') {
+        return new PatternMatcher(toRegexPattern(globPattern));
+      }
+    }
+
+    return new StringMatcher(globPattern);
+  }
+
+  private static Pattern toRegexPattern(String globPattern) {
+    int tokenStart = -1;
+    StringBuilder patternBuilder = new StringBuilder();
+    for (int i = 0; i < globPattern.length(); i++) {
+      char c = globPattern.charAt(i);
+      if (c == '*' || c == '?') {
+        if (tokenStart != -1) {
+          patternBuilder.append(Pattern.quote(globPattern.substring(tokenStart, i)));
+          tokenStart = -1;
+        }
+        if (c == '*') {
+          patternBuilder.append(".*");
+        } else {
+          // c == '?'
+          patternBuilder.append(".");
+        }
+      } else {
+        if (tokenStart == -1) {
+          tokenStart = i;
+        }
+      }
+    }
+    if (tokenStart != -1) {
+      patternBuilder.append(Pattern.quote(globPattern.substring(tokenStart)));
+    }
+    return Pattern.compile(patternBuilder.toString());
```

Comment:
This may blow with `PatternSyntaxException`. I think this path should be protected and most probably log warning / return "FalseMatcher".

Reply:
Do you think it's possible even though we only add quoted strings and these simple wildcards?

Reply:
Good question! I thought it can be broken by nonsensical pattern (like a sequence of dots) but it seems that Java allows that without question. So now it should be always safe, at least as far as I know regexes ;)